lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -91,33 +51,15 @@ def test_missing_keys(
         Note that "start", and "end" are required whereas "user" is not mandatory;
         however, Pbench will silently ignore any additional keys that are
         specified.
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json=keys,
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        missing = [k for k in ("start", "end") if k not in keys]
-        assert (
-            response.json.get("message")
-            == f"Missing required parameters: {','.join(missing)}"
-        )
-
-    def test_bad_dates(self, client, server_config, user_ok, pbench_token):
-        """
-        Test behavior when a bad date string is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": "drb", "start": "2020-12", "end": "2020-19"},
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Value '2020-19' (str) cannot be parsed as a date/time string"
-        )
+       """
+        self.required_keys = [
+            p[0]
+            for p in ControllersList(
+                client.config, client.logger
+            ).schema.parameters.items()
+            if p[1].required
+        ]
+        super().test_missing_keys(client, server_config, keys, user_ok, pbench_token)
```

Comment:
Having one test call another strikes me as being problematic, even if the other is the corresponding superclass method.  I can hope that PyTest's introspection is clever enough to properly note the override and not call the superclass method directly calling the subclass overrider instead.

Nevertheless, I think it is your intention that every subclass _must_ override `test_missing_keys()`, in which case the superclass function devolves to being just a helper function; therefore you should name it without the `test_` prefix, so that there is no ambiguity about whether it is a test.